src/pybind/mgr/cephadm/module.py

```diff
@@ -1037,8 +1034,65 @@ def create_osds(self, drive_group):
         return self.get_hosts().then(lambda hosts: self._create_osd(hosts, drive_group))
 
     @with_services('osd')
-    def remove_osds(self, osd_ids, services):
-        # type: (List[str], List[orchestrator.ServiceDescription]) -> AsyncCompletion
+    def remove_osds(self, osd_ids, destroy=False, services=[]):
+        # type: (List[str], bool, List[orchestrator.ServiceDescription]) -> AsyncCompletion
+
+        def ok_to_destroy():
+            return self.mon_command({
+                'prefx': 'osd safe-to-destroy',
+                'ids': osd_ids,
+            })
+        if destroy:
+            ret = ok_to_destroy()
+            if ret.retval:
+                raise OrchestratorValidationError('osd_ids not safe to destroy: {} {}'.format(osd_ids, ret.stderr))
+            for osd_id in osd_ids:
+                ret = self.mon_command({
+                    'prefx': 'osd destroy',
+                    'id': osd_id,
+                    'yes_i_really_mean_it': True
+                })
+                if ret.retval:
+                    raise ValueError(
+                        'osd destroy failed: {} {}'.format(osd_id, ret.stderr))
+        else:
+            for osd_id in osd_ids:
+                ret = self.mon_command({
+                    'prefx': 'osd crush reweight',
+                    'name': osd_id,
```

Comment:
```suggestion
                    'name': f'osd.{osd_id}',
```

`osd crush reweight` expects the full name as registered in the crush map.